### DISCUSSION TITLE
Custom build step with @now/python?

### DISCUSSION THREAD

**Discussion by simonw** [vercel/vercel#4055] [General] [answered]
If I'm using the `@now/python` builder is there a way for me to run some custom code during the build phase?

I'd like to do things like generate static assets to include in the build.

**Comment by styfle** [comment, marked as answer]
Hi @simonw 

You can run configure a Build Command and Output Directory using [Project Settings](https://zeit.co/blog/advanced-project-settings). All the files in the Output Directory are considered static files and served with the correct mime type.

If you have a static site generator written in javascript, you can add a `package.json` to the root like so:

```json
{
  "private": true,
  "scripts": {
    "build": "invoke your ssg cli here"
  }
}
```

This assumes the SSG will write the output to the `public` directory.

**Comment by simonw** [comment]
Is there a way to specify a custom build command from within the `now.json` file?

I always prefer having this kind of configuration in source control along with the rest of my code.

**Reply by styfle** [reply]
That is not available at this time.

**Reply by simonw** [reply]
Please consider this a feature request then! One of the things that drew me to Now in the first place was that my configuration could easily live in my repo and I wouldn't have to worry about clicking around in configuration pages (like I so often have to do with AWS).

**Reply by styfle** [reply]
Thanks, I'll pass on your feedback to the team!

**Comment by styfle** [comment]
@simonw I just thought of a possible solution for your Datasette project.

Today, we [detect many static site generators](https://vercel.com/docs/v2/build-step#build-&-development-settings) such as Next.js, Jekyll, Hugo, Gatsby, etc.

Is there a way we can detect if a project is using Datasette? That way we can automatically populate the Build Command and Output Directory. Perhaps there is a dependency in `requirements.txt` we could look for?

**Reply by simonw** [reply]
Oh we could definitely do something like that. I'm actually thinking about making a change to the core project that would make that easier - but right now if `datasette` is listed in the `requirements.txt` and there's at least one `*.db` file that could be a trigger.

**Reply by simonw** [reply]
Here are notes on that core project change I'm considering which could fit in with the idea of automatically detecting a Datasette project: https://github.com/simonw/datasette/issues/731

**Comment by leo** [comment]
Additionally, if that's not possible, what do you think about a `vercel-build.sh` file that would always be invoked and could provide the command.

**Reply by simonw** [reply]
That would work really well for me.

**Comment by tianhuil** [comment]
I haven't gotten it to work but perhaps this is a solution: https://github.com/almapario/vercel-poetry
